src/MacroTools/QuestSystem/UtilityStructs/ObjectiveUnitIsDead.cs

```diff
@@ -0,0 +1,43 @@
+﻿using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using WCSharp.Events;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.QuestSystem.UtilityStructs
+{
+  public class ObjectiveUnitIsDead : Objective
+  {
+    public ObjectiveUnitIsDead(unit unitToKill)
+    {
+      trigger trig = CreateTrigger();
```

Comment:
Up to you, but you can use Linq-style syntax if you want. E.g:

```cs
CreateTrigger()
   .RegisterUnitEvent(unitToKill, EVENT_UNIT_DEATH)
   .AddAction(trig, OnUnitDeath)
```

Reply:
I'm not as used to using Linq-style syntax but makes sense and one to keep in mind.